dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -208,12 +208,14 @@ public AuthenticationJourneyType GetJourneyType()
             return AuthenticationJourneyType.StaffSignIn;
         }
 
+#pragma warning disable CS0618 // Type or member is obsolete
         if (TryGetOAuthState(out var oAuthState) &&
             oAuthState.RequiresTrnLookup &&
-            oAuthState.TrnRequirementType == TrnRequirementType.Legacy)
+            (oAuthState.TrnRequirementType == TrnRequirementType.Legacy || oAuthState.HasScope(CustomScopes.Trn)))
```

Comment:
Modified this subtly so that a single client can access both the new and legacy journeys by specifying the `dqt:read` or `trn` scope, respectively.